apps/freeablo/faworld/playerfactory.cpp

```diff
@@ -6,59 +6,56 @@
 namespace FAWorld
 {
 
-PlayerFactory::PlayerFactory(DiabloExe::DiabloExe& exe)
+PlayerFactory::PlayerFactory(const DiabloExe::DiabloExe& exe)
     : mExe(exe)
 {
 }
 
-Player* PlayerFactory::create(const std::string& playerClass)
+Player* PlayerFactory::create(const std::string& playerClass) const
 {
-    mPlayer = new Player();
-    mCharStats = mExe.getCharacterStat(playerClass);
+    auto player = new Player();
+    auto charStats = mExe.getCharacterStat(playerClass);
+
+    ActorStats* stats = NULL;
 
     if(playerClass == "Warrior")
-    {
-        createWarrior();
-    }
+        stats = createWarrior(player, charStats);
     else if(playerClass == "Rogue")
-    {
-        createRogue();
-    }
+        stats = createRogue(player, charStats);
     else
-    {
-        createSorcerer();
-    }
+        stats = createSorcerer(player, charStats);
+
+    player->setStats(stats);
+    stats->setActor(player);
 
-    mPlayer->setStats(mStats);
-    mStats->setActor(mPlayer);
-    mPlayer->mInventory.collectEffects();
-    mStats->recalculateDerivedStats();
+    player->mInventory.collectEffects();
+    stats->recalculateDerivedStats();
 
-    return mPlayer;
+    return player;
 }
 
-void PlayerFactory::createWarrior()
+ActorStats* PlayerFactory::createWarrior(Player* player, const DiabloExe::CharacterStats& charStats) const
```

Comment:
Would be more natural if createWarrior/createRogue/createSorcerer returns object of class Player rather than ActorStats. Maybe try to remove parameter "player", create ActorStats inside this func and return Player.